### ISSUE TITLE
RegexGenerator fixer hijacks formatting

### ISSUE THREAD

**Issue by danielchalmers** [dotnet/runtime#71852]
**Version Used**: Visual Studio 17.3P2, 7.0.100-preview.5.22307.18

**Steps to Reproduce**:

With code:

```csharp
public static string CollapseWhitespace(this string text) =>
        Regex.Replace(text, @"\s+", " ");
```

Choose `Convert to 'RegexGenerator'`.
 

**Expected Behavior**:

Newline preserved after lambda.

```csharp
public static string CollapseWhitespace(this string text) =>
        MyRegex().Replace(text, " ");
```

**Actual Behavior**:

Newline removed after lambda.

```csharp
public static string CollapseWhitespace(this string text) => MyRegex().Replace(text, " ");
```

**Comment by joperezr** [comment]
Yes, this is our code fixer for the Regex Generator. @CyrusNajmabadi we are just currently calling the Roslyn APIs to replace the node that had the instantiation of Regex and swapping it for an invocation instead, is there anything we are missing in order to preserve formatting?

**Comment by CyrusNajmabadi** [comment]
can you link me to your fixer code?

**Comment by CyrusNajmabadi** [comment]
After this line, try adding the following:

https://github.com/dotnet/runtime/blob/cae954add0f01c85de7fef8cc41c2a11d1c920e1/src/libraries/System.Text.RegularExpressions/gen/UpgradeToRegexGeneratorCodeFixer.cs#L178

```c#
method = method.WithLeadingTrivia(nodeToFix.GetLeadingTrivia()).WithTrailingTrivia(nodeToFix.GetTrailingTrivia());
```
